app/src/androidTest/java/com/android/unio/ui/authentication/AccountDetailsTest.kt

```diff
@@ -70,105 +72,128 @@ class AccountDetailsTest {
-    composeTestRule.onNodeWithTag("AccountDetailsFirstNameTextField").performTextInput("John")
-    composeTestRule.onNodeWithTag("AccountDetailsLastNameTextField").performTextInput("Doe")
-    composeTestRule.onNodeWithTag("AccountDetailsBioTextField").performTextInput("I am a student")
+    composeTestRule
+        .onNodeWithTag(AccountDetailsTestTags.FIRST_NAME_TEXT_FIELD)
+        .performTextInput("John")
+    composeTestRule
+        .onNodeWithTag(AccountDetailsTestTags.LAST_NAME_TEXT_FIELD)
+        .performTextInput("Doe")
+    composeTestRule
+        .onNodeWithTag(AccountDetailsTestTags.BIOGRAPHY_TEXT_FIELD)
+        .performTextInput("I am a student")
 
-    composeTestRule.onNodeWithTag("AccountDetailsFirstNameTextField").assertTextContains("John")
-    composeTestRule.onNodeWithTag("AccountDetailsLastNameTextField").assertTextContains("Doe")
-    composeTestRule.onNodeWithTag("AccountDetailsBioTextField").assertTextContains("I am a student")
+    composeTestRule
+        .onNodeWithTag(AccountDetailsTestTags.FIRST_NAME_TEXT_FIELD)
+        .assertTextContains("John")
+    composeTestRule
+        .onNodeWithTag(AccountDetailsTestTags.LAST_NAME_TEXT_FIELD)
+        .assertTextContains("Doe")
+    composeTestRule
+        .onNodeWithTag(AccountDetailsTestTags.BIOGRAPHY_TEXT_FIELD)
+        .assertTextContains("I am a student")
   }
 
   @Test
   fun testInterestsButtonWorksCorrectly() {
-    composeTestRule.onNodeWithTag("AccountDetailsInterestsButton").performScrollTo().performClick()
-    composeTestRule.onNodeWithTag("InterestOverlayTitle").assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(AccountDetailsTestTags.INTERESTS_BUTTON)
+        .performScrollTo()
+        .performClick()
+    composeTestRule.onNodeWithTag(InterestsOverlayTestTags.TITLE_TEXT).assertIsDisplayed()
   }
 
   @Test
   fun testSocialsButtonWorksCorrectly() {
-    composeTestRule.onNodeWithTag("AccountDetailsSocialsButton").performScrollTo().performClick()
-    composeTestRule.onNodeWithTag("SocialOverlayTitle").assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(AccountDetailsTestTags.SOCIALS_BUTTON)
+        .performScrollTo()
+        .performClick()
+    composeTestRule.onNodeWithTag(SocialsOverlayTestTags.TITLE_TEXT).assertIsDisplayed()
   }
 
   @Test
   fun testAddingInterestsCorrectlyModifiesTheFlowRow() {
-    composeTestRule.onNodeWithTag("AccountDetailsInterestsButton").performClick()
-    composeTestRule.onNodeWithTag("InterestOverlayClickableRow: 0").performClick()
-    composeTestRule.onNodeWithTag("InterestOverlayClickableRow: 1").performClick()
-    composeTestRule.onNodeWithTag("InterestOverlaySaveButton").performClick()
+    composeTestRule.onNodeWithTag(AccountDetailsTestTags.INTERESTS_BUTTON).performClick()
+    composeTestRule.onNodeWithTag(InterestsOverlayTestTags.CLICKABLE_ROW + "0").performClick()
+    composeTestRule.onNodeWithTag(InterestsOverlayTestTags.CLICKABLE_ROW + "1").performClick()
```

Comment:
I'm maybe pushing a bit too far, but the "0" and "1" could also be stored in a constant.